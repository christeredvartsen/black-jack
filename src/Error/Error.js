import React from 'react';
import PropTypes from 'prop-types';
import RestartGame from '../RestartGame/RestartGame';
import './Error.scss';

const Error = ({ message, reset }) => (
  <React.Fragment>
    <p>An error occurred: <span className="errorMessage">{message}</span>.</p>
    <RestartGame reset={reset} />
  </React.Fragment>
);
Error.propTypes = {
  message: PropTypes.string.isRequired,
  reset: PropTypes.func.isRequired,
};

export default Error;