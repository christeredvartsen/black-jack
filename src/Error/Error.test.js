import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Error from './Error';

describe('<Error />', () => {
  test('matches snapshot', () => {
    const tree = shallow(
      <Error message="some error message" reset={jest.fn()} />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});