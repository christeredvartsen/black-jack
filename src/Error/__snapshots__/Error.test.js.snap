// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> matches snapshot 1`] = `
<Fragment>
  <p>
    An error occurred: 
    <span
      className="errorMessage"
    >
      some error message
    </span>
    .
  </p>
  <RestartGame
    reset={[MockFunction]}
  />
</Fragment>
`;
